crates/oxc_minifier/src/ast_passes/peephole_fold_constants.rs

```diff
@@ -234,8 +231,9 @@ impl<'a> PeepholeFoldConstants {
                         UnaryOperator::UnaryNegation if un.argument.is_number() => {
                             // `-~1` -> `2`
                             if let Expression::NumericLiteral(n) = &mut un.argument {
-                                is_within_i32_range(n.value).then(|| {
-                                    let value = !(-n.value.to_js_int_32());
+                                is_valid(n.value).then(|| {
+                                    // Here, we need to handle the exceeded scenarios with C.
```

Comment:
> Here, we need to handle the exceeded scenarios with C.

This comment doesn't explain anything. JavaScript, C, and Rust all follow the same number system.

Reply:
In the event of an overflow, we should handle the scenario as described in C and release the optimized version. In debug cases, it may potentially panic.